content/news/2023-10-21-bevy-0.12/index.md

```diff
@@ -19,6 +19,319 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
 
 <div class="release-feature-authors">authors: @author</div>
 
+## Automatic Batching and Instancing, and the Road to GPU-driven Rendering
+
+<div class="release-feature-authors">authors: Rob Swain (@superdump), @james-j-obrien, @JMS55, @inodentry, @robtfm, @nicopap, @teoxoy, @IceSentry, @Elabajaba</div>
+
+Bevy's renderer performance for 2D and 3D meshes can improve a lot. Both CPU and graphics API / GPU bottlenecks can be removed to give significantly higher frame rates. As always with Bevy, we want to make the most of the platforms you use, from the constraints of WebGL2 and mobile devices, to the highest-end native discrete graphics cards. A solid foundation can support all of this.
+
+### What are the bottlenecks?
+
+One major bottleneck is the structure of the data used for rendering.
+
+* Mesh entity data is stored in one uniform buffer, but has to be rebound at different dynamic offsets for every single draw.
+* Material type data (e.g. `StandardMaterial` uniform properties, but not textures) are stored in individual uniform buffers that have to be rebound per draw if the material changes.
+* Material textures are stored individually and have to be rebound per draw if a mesh texture changes.
+* Mesh index / vertex buffers are stored individually per-mesh and have to be rebound per draw.
```

Comment:
This doesn't relate to work that was done this release. I think we should cut it.